java.lexer/src/org/netbeans/lib/java/lexer/JavaLexer.java

```diff
@@ -1297,6 +1306,25 @@ private int translateSurrogates(int c) {
             JavaTokenId.LINE_COMMENT, JavaTokenId.WHITESPACE
     );
 
+    private Integer getVersionAsInt(String version) {
+        Integer ver = null;
+        if (version != null) {
+            try {
```

Comment:
There is [SpecificationVersion](http://bits.netbeans.org/dev/javadoc/org-openide-modules/org/openide/modules/SpecificationVersion.html#SpecificationVersion-java.lang.String-) which you might fine useful in this situation.

Reply:
Hi, JavaLexer expects Integer version, so this method converts String 1.x to x, eg 1.8 to 8. SpecificationVersion is useful for comparing versions, not sure if I can use it here.